service/internal/podcidr/pod_cidr.go

```diff
@@ -72,18 +70,30 @@ func (p *PodCIDR) lookupCluster(ctx context.Context, cr metav1.Object) (bootstra
 		ctx,
 		&list,
 		client.InNamespace(cr.GetNamespace()),
-		client.MatchingLabels{label.Cluster: key.ClusterID(cr)},
```

Comment:
Could we not change the value of label.Cluster to cluster.x-k8s.io/cluster-name instead?

Reply:
Because my kubeadmconfigs do not have control-plane in their names

Reply:
We could, but still there are multiple `KubeadmConfig`s with that label in the namespace.

Reply:
Could we not get what we need from teh kubeadmcontrolplane instead?

Reply:
Does `KubeadmControlPlane` contain fields analogous to `.spec.clusterConfiguration.networking.podSubnet`? I could not find one.

Reply:
You are right. Could not then get the kubeadmconfigs with the control plane label with a name starting by the cluster id? In that case, could we change the condition to allow any number of kubeadm config matching this?

Reply:
Can we match here on the CAPI label for clusters instead? `cluster.x-k8s.io/cluster-name: CLUSTERID`?

Maybe I am missing something obvious

Reply:
Sure, why not. I still need to filter by name though:
```
~ k get kubeadmconfigs.bootstrap.cluster.x-k8s.io -n antonia --show-labels
NAME                          AGE    LABELS
antonia-control-plane-8cd89   122m   cluster.x-k8s.io/cluster-name=antonia,cluster.x-k8s.io/control-plane=,cluster.x-k8s.io/watch-filter=0.0.1
antonia-md-0-ldcfl            130m   cluster.x-k8s.io/cluster-name=antonia,cluster.x-k8s.io/deployment-name=antonia-md-0,cluster.x-k8s.io/watch-filter=0.0.1,machine-template-hash=565822541
```

Reply:
Can you not also add `cluster.x-k8s.io/control-plane=` in the matchingLabels ?
The control-plane in the name only applies to CAPA. This would not work for CAPV

Reply:
Hmm, maybe I could actually filter by `cluster.x-k8s.io/control-plane=`

Reply:
Why does the control-plane label not exist in CAPV?

Reply:
The control plane label exist, but the previous code was filtering by the kubeadmconfig name which does not contain control-plane in capv.

antonia-control-plane-8cd89 is not the name of the generated one for cavp, it is just antonia-8cd89

Reply:
The inconsistency is killing me. But okay I get it :+1:

Reply:
You should have faith in labels :D

Reply:
@QuentinBisson do you foresee any issues with VMware and current label filtering? It no longer pays mind to config name.

Reply:
I like this label approach way more, especially since we have kyverno adding the labels for us :D